File: app/src/main/java/com/wiktor/weater1/newWeaher/weatherDetalisation/mvp/WeatherView.java
package com.wiktor.weater1.newWeaher.weatherDetalisation.mvp;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.wiktor.weater1.newWeaher.cityList.model.CityModel;
import com.wiktor.weater1.newWeaher.weatherDetalisation.model.WeatherModelForView;

import java.util.List;

public interface WeatherView extends MvpView {

    @StateStrategyType(AddToEndStrategy.class)
    void showData(double temperat);

    @StateStrategyType(AddToEndStrategy.class)
    void showErrorDialog(String error);

    @StateStrategyType(AddToEndStrategy.class)
    void showList(List <WeatherModelForView> list);

    @StateStrategyType(AddToEndStrategy.class)
    void changeTitle(CityModel cityModel);

    @StateStrategyType(AddToEndStrategy.class)
    void hideProgressBar();
}

File: app/src/main/java/com/wiktor/weater1/newWeaher/weatherDetalisation/mvp/WeatherView.kt
package com.wiktor.weater1.newWeaher.weatherDetalisation.mvp

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.wiktor.weater1.newWeaher.cityList.model.CityModel
import com.wiktor.weater1.newWeaher.weatherDetalisation.model.WeatherModelForView

interface WeatherView : MvpView {

    @StateStrategyType(AddToEndStrategy::class)
    fun showData(temperat: Double)

    @StateStrategyType(AddToEndStrategy::class)
    fun showErrorDialog(error: String)

    @StateStrategyType(AddToEndStrategy::class)
    fun showList(list: List<WeatherModelForView>)

    @StateStrategyType(AddToEndStrategy::class)
    fun changeTitle(cityModel: CityModel)

    @StateStrategyType(AddToEndStrategy::class)
    fun hideProgressBar()
}